features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/internal/FlagsFeature.kt

```diff
@@ -0,0 +1,217 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.internal
+
+import android.content.Context
+import android.content.pm.ApplicationInfo
+import android.util.Log
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.feature.Feature.Companion.FLAGS_FEATURE_NAME
+import com.datadog.android.api.feature.Feature.Companion.RUM_FEATURE_NAME
+import com.datadog.android.api.feature.FeatureContextUpdateReceiver
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.feature.StorageBackedFeature
+import com.datadog.android.api.storage.FeatureStorageConfiguration
+import com.datadog.android.flags.FlagsClient
+import com.datadog.android.flags.FlagsConfiguration
+import com.datadog.android.flags.internal.net.ExposuresRequestFactory
+import com.datadog.android.flags.internal.net.PrecomputedAssignmentsRequestFactory
+import com.datadog.android.flags.internal.storage.ExposureEventRecordWriter
+import com.datadog.android.flags.internal.storage.NoOpRecordWriter
+import com.datadog.android.flags.internal.storage.RecordWriter
+import com.datadog.android.log.LogAttributes.RUM_APPLICATION_ID
+
+/**
+ * Type alias for a function that logs a message with a given level.
+ *
+ * Used to bridge between module components and a graceful policy-enabled logger.
+ */
+internal typealias LogWithPolicy = (String, InternalLogger.Level) -> Unit
+
+internal class FlagsFeature(private val sdkCore: FeatureSdkCore, internal val flagsConfiguration: FlagsConfiguration) :
+    StorageBackedFeature,
+    FeatureContextUpdateReceiver {
+
+    @Volatile
+    internal var applicationId: String? = null
+
+    @Volatile
+    internal var processor: EventsProcessor = NoOpEventsProcessor()
+
+    @Volatile
+    internal var dataWriter: RecordWriter = NoOpRecordWriter()
+
+    @Volatile
+    private var isInitialized = false
+
+    /**
+     * Indicates whether the app is running in debug mode.
+     * Set once during onInitialize() and never changes for the process lifetime.
+     */
+    @Volatile
+    private var isDebugBuild: Boolean = false
+
+    // region Domain Objects
+
+    /**
+     * This is the same as the default configuration except
+     * that we limit to 50 items per batch, the same as the JS library does.
+     */
+    override val storageConfiguration =
+        FeatureStorageConfiguration.DEFAULT.copy(
+            maxItemsPerBatch = MAX_ITEMS_PER_BATCH
```

Comment:
is this intake limitation?

Reply:
no. The intake limit is 5mB or around 22k records. Chatting with the evp owner, we're thinking maybe 500 is a more reasonable default.